.changeset/four-games-walk.md

```diff
@@ -0,0 +1,12 @@
+---
+'@mastra/core': major
+---
+
+The `@mastra/core` package no longer allows top-level imports. You must use subpath imports for all imports.
+
+For example:
+
+```diff
+- import { Mastra } from "@mastra/core";
```

Comment:
this is the only import that i would allow from main :) 

`import { Mastra } from "@mastra/core";` should still be possible

Reply:
Okay 👍

Reply:
@wardpeet done